vmcnet/updates/proxsr.py

```diff
@@ -0,0 +1,154 @@
+# >>>>>>>>>> from gg-min-sr-mom >>>>>>>>>>
+
+"""Stochastic reconfiguration (SR) routine."""
+import jax
+import jax.flatten_util
+import jax.numpy as jnp
+
+from vmcnet.utils.typing import Array, ModelApply, P, Tuple
+
+import chex
+from vmcnet.utils.pytree_helpers import (
+    multiply_tree_by_scalar,
+    tree_inner_product,
+)
+from vmcnet import utils
+
+
+def get_fisher_inverse_fn(
+    log_psi_apply: ModelApply[P],
+    damping_type: str = "diag_shift",
+    damping: chex.Scalar = 0.001,
+    complement_decay: chex.Scalar = 0.95,
+):
+    """Get a Fisher-preconditioned update.
+
+    Given a gradient update grad_E, the function returned here approximates
```

Comment:
Can you clean up this comment? This is still describing the generic SR algorithm rather than ProxSR. You can just remove most of the discussion if you don't want to describe things in detail like it does now.